sycl/doc/extensions/LevelZeroBackend/LevelZeroBackend.md

```diff
@@ -293,6 +312,25 @@ should take ownership of the passed native handle. The default behavior is to
 transfer the ownership to the SYCL runtime. See section 4.4 for details. If
 the behavior is "transfer" then the runtime is going to destroy the input
 Level-Zero kernel</td>
+</tr><tr>
+<td>
+
+``` C++
+template <backend Backend, typename T, int Dimensions = 1,
+          typename AllocatorT = buffer_allocator>
+typename std::enable_if<Backend == backend::ext_oneapi_level_zero,
+                        buffer<T, Dimensions, AllocatorT>>::type
+make_buffer(
+    const backend_input_t<backend::ext_oneapi_level_zero,
+                          buffer<T, Dimensions, AllocatorT>> &BackendObject,
+    const context &TargetContext, event AvailableEvent = {})
+```
+</td>
+<td>Constructs a SYCL buffer instance from a Level-Zero memory allocation <code>void *</code>. 
+The <code>Context</code> argument must be a valid SYCL context encapsulating a Level-Zero context, and the Level-Zero memory must be allocated on the same context. 
+The <code>AvailableEvent</code> argument must be a valid SYCL event, the instance of the SYCL buffer class template being constructed must wait for the SYCL event parameter, signaled event means that the memory native handle is ready to be used.
+The <code>Size</code> input structure member specifies a size of a Level-Zero memory allocation in bytes.

```

Comment:
Why do we have to know the size? Is it only to report it with piMemGetInfo? Maybe we should just report "unknown" value for interop buffer, since the size should be known to the application already

Reply:
To construct sycl::buffer object we need to know the size, sycl::buffer can't be constructed with unknown size. For OpenCL size can be queried from cl_mem using clMemGetInfo but for Level Zero native memory handle is just a raw pointer void*. The only option I see is to provide the size.

Reply:
But why do you think size is required at construction? What happens if you let the size to be some fake value decoded as unknown?

Reply:
I have a separate question about the size.  Must the size be the same value that was passed to `zeMemAllocShared` / `zeMemAllocDevice` / `zeMemAllocHost`, or is it legal to pass a smaller value?  One could imagine that it might be useful to allow a smaller value in some cases.  Either way, the spec should clarify what the requirement is.

Reply:
> But why do you think size is required at construction? What happens if you let the size to be some fake value decoded as unknown?
Problem is that if size is unknown then we will not be able to create a host accessor to the buffer if someone provided a pointer created with zeMemAllocDevice. In this case we need to allocate a host buffer and copy device allocation to this host buffer. And we need to know the size of a buffer to allocate a host buffer.

We can disallow zeMemAllocDevice pointers to be provided to make_buffer or disallow creation of the host buffers but I'm not sure if it is an option.

Also get_size, get_range methods of the buffer will return some unknown value.

Reply:
It looks like we can query size of the allocation: https://spec.oneapi.io/level-zero/latest/core/api.html?highlight=getmemory#zememgetaddressrange
I will check if it works and will drop the Size from make_buffer.

Reply:
Unfortunately, zeMemGetAddressRange doesn't return user-provided buffer size but some internal value. That's why I need to keep Size parameter to be able to handle buffers allocated on device which need shadow host allocation in case of copy back.


> I have a separate question about the size.  Must the size be the same value that was passed to `zeMemAllocShared` / `zeMemAllocDevice` / `zeMemAllocHost`, or is it legal to pass a smaller value?  One could imagine that it might be useful to allow a smaller value in some cases.  Either way, the spec should clarify what the requirement is.
@gmlueck I have specified in the doc that size must have the same value that was passed to  `zeMemAllocShared` / `zeMemAllocDevice` / `zeMemAllocHost`. If smaller buffer is needed then user should create sub-buffer.